plugins/source/aws/scripts/policy_docs/main.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+)
+
+//go:embed templates/*.go.tpl
+var templatesFS embed.FS
+
+var reInline = regexp.MustCompile(`(?i)\\ir (.+)`)
+var reTable = regexp.MustCompile(`(?i)(?:from|join)\s+(\w+)`)
+var reTitle = regexp.MustCompile(`(?i)\'(.+)\'\s+as\s+title,`)
+var reIsPolicyQuery = regexp.MustCompile(`(?i)insert\s+into\s+aws_policy_results`)
+
+type Index struct {
+	Policies []Policy `json:"policies"`
+}
+
+type Policy struct {
+	Name string `json:"name"`
+	Path string `json:"path"`
+}
+
+func readIndex(dir string) (*Index, error) {
+	indexFile := path.Join(dir, "index.json")
+	content, err := os.ReadFile(indexFile)
+	if err != nil {
+		return nil, fmt.Errorf("error opening file %s: %w", indexFile, err)
+	}
+	var index Index
+	err = json.Unmarshal(content, &index)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal json from index file %s: %w", indexFile, err)
+	}
+	return &index, nil
+}
+
+type PolicyInfo struct {
+	Name    string
+	Queries []Query
+}
+
+// Tables returns the unique set of tables needed to run all queries
+func (pi PolicyInfo) Tables() []string {
+	t := map[string]struct{}{}
+	for _, q := range pi.Queries {
+		for _, table := range q.Tables {
+			t[table] = struct{}{}
+		}
+	}
+	var final []string
+	for k := range t {
+		final = append(final, k)
+	}
+	sort.Strings(final)
+	return final
+}
+
+type Query struct {
+	Title  string
+	Path   string
+	Tables []string
+}
+
+func getPolicyInfo(prefix, dir string, policy Policy) (*PolicyInfo, error) {
+	queries, err := extractQueries(prefix, path.Join(dir, policy.Path))
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract queries: %w", err)
+	}
+	return &PolicyInfo{
+		Name:    policy.Name,
+		Queries: queries,
+	}, nil
+}
+
+func extractQueries(prefix, sqlPath string) ([]Query, error) {
+	b, err := os.ReadFile(sqlPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file %v: %w", sqlPath, err)
+	}
+	content := string(b)
+	var queries []Query
+	tableMatches := reTable.FindAllStringSubmatch(content, -1)
+	isPolicyQuery := reIsPolicyQuery.MatchString(content)
+	if isPolicyQuery && len(tableMatches) > 0 {
+		q := Query{}
+		for _, m := range tableMatches {
+			if !strings.HasPrefix(m[1], prefix) {
+				continue
+			}
+			q.Tables = append(q.Tables, m[1])
+			q.Path = sqlPath
+		}
+
+		titleMatches := reTitle.FindAllStringSubmatch(content, -1)
+		if len(titleMatches) == 0 {
+			log.Printf("WARN: Failed to find title for query in %v", sqlPath)
```

Comment:
Nice log level :)

Reply:
Haha, yeah, this is a bit WIP. I think we should probably fail the script if this happens, so that it needs to be fixed in the query first.